### PR TITLE
Feature: Add named chunk for `import()` by specified comment format.

### PR DESCRIPTION
**Pull request by mc-zone**
**What kind of change does this PR introduce?**

Feature

**Did you add tests for your changes?**

Yes. Added tests in `test/cases/chunks/named-chunks/` .

**If relevant, link to documentation update:**

N/A

**Summary**

**motivation**

Support specify chunk name for `import()` by following specified comment block after the param:

```javascript
import('./foo' /* webpackChunkName = "myChunkName" */).then(function(foo){
    //...  
});
```

Thus we can use chunk name like `requre.ensure` and avoid conflicts with the specification. 

The above is like following when using `require.ensure`:

```javascript
require.ensure([], function(require){
  var foo = require("./foo");
}, "myChunkName");
```

** open issue for more information. **

#1949 .

**Does this PR introduce a breaking change?**

No. It is a new feature supported by add new param/syntax.

**Other information**

### CODE REVIEW

File: test/cases/chunks/named-chunks/index.js [hunk 1]
```diff
@@ -33,3 +33,46 @@ it("should handle empty named chunks", function(done) {
 		sync = false;
 	});
 });
+
+it("should be able to use named chunks in import()", function(done) {
+	var sync = false;
+  import("./empty?e" /* webpackChunkName = "import-named-chunk" */).then(function(result){
```
Comment (jhnns) [inline test/cases/chunks/named-chunks/index.js]: What happened to the indentation? :)

**Comment by jhnns** [comment]
Thank you 👍
That's a useful change!

@sokra opinions? 😁

File: test/cases/chunks/named-chunks/index.js (hunk 1 above)
Comment (mc-zone) [inline test/cases/chunks/named-chunks/index.js]: My bad. Looks like didn't load the .editorconfig at the time. It's weird. I've reinstall my editor plugin then repaired it.

**Comment by roastlechon** [comment]
love it!

**Comment by hulkish** [comment]
My name is not Donald Trump, and I support this PR.

**Comment by sokra** [comment]
I'm generally 👍, it just need some changes. I'll do this in a few days.

**Comment by mc-zone** [comment]
Awesome ! :+1: :beer:

**Comment by sokra** [comment]
I'll first merge #4635 than rebase this again and after this it can be merged 🎉

**Comment by sokra** [comment]
Thanks

**Comment by roastlechon** [comment]
@sokra when will a release be cut for this feature?

**Comment by victoryancn** [comment]
super cool, save me a lot of time ,thanks!

**Comment by gztchan** [comment]
I suggest that this feature should be put into the docs.

**Comment by mc-zone** [comment]
> I suggest that this feature should be put into the docs.

There are already the docs: https://webpack.js.org/guides/code-splitting-async/#chunk-names

**Comment by mieszko4** [comment]
It would be great to be able to use variables:
```es
import(/* webpackChunkName: `lang-${lang}` */ `./i18n/locales/${lang}.json`)
```

Or at least to be able to use original `[id]` of the chunk name so that chunks could be prefixed:
```es
import(/* webpackChunkName: `lang-[id]` */ `../../../i18n/locales/${lang}.json`)
```

**Comment by liqiang372** [comment]
How to use `import()` to load several files ?
in `require.ensure` we can

```javascript
require.ensure([], function(){
  var a = require('./a');;
  var b = require('./b');
  b(a()); // let them work together
})
```
How to acheive this using `import()`

```javascript
import('./a').then(a) {
  import('./b').then(b) {
    b(a()) // ???
  }
}

**Comment by hulkish** [comment]
like this:

```js
Promise.all([
  import('./a'),
  import('./b'),
  import('./c')
]).then(([a, b, c]) => { 
  // do something with a, b or c
});
```

**Comment by alex-shamshurin** [comment]
I see this works only for `import(/* webpackChunkName: "a" */ './a'),` but not for `import( './a', /* webpackChunkName: "a" */),`.

**Comment by niieani** [comment]
@alex-shamshurin you mean with a comma after the first parameter? If you delete the comma, it will work both ways.

**Comment by alex-shamshurin** [comment]
Yep, but Prettier place it always outside brackets, on next line. Could it work with it? I always have to replace it manually after formatting.

**Comment by mc-zone** [comment]
> Yep, but Prettier place it always outside brackets, on next line. Could it work with it? I always have to replace it manually after formatting.

@alex-shamshurin Do you have any specific config with Prettier? It seems I can't reproduce it on [https://prettier.io/playground](https://prettier.io/playground/#%7B%22content%22%3A%22import(%20'.%2Fa'%20%2F*%20webpackChunkName%3A%20%5C%22a%5C%22%20*%2F)%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D).

**Pull request merged**